### PR TITLE
Add SocketHandler

### PR DESCRIPTION
**Pull request by aminalaee**
From CPython [here](https://github.com/python/cpython/blob/9d09e7b0263c248659f3b4a2c597fca9ac4f8f91/Lib/logging/handlers.py#L525).

- The tests would be cleaner if they were using classes but we would lose `pytest` features like `repeat` and `fixtures`
- I specifically didn't add the benchmarks as they would need extra work for the server. Maybe we can add them in a separate PR.

### CODE REVIEW

File: src/picologging/__init__.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
     _checkLevel,
     StringTemplateStyle,
     BufferingFormatter,
+    makeLogRecord,
```
Comment (tonybaloney) [inline src/picologging/__init__.py:29]: Why is `makeLogRecord()` imported? That function shouldn't be called anywhere as it doesn't make a `LogRecord` instance from the pico logging module, it's also really slow (and one of the reasons pico logging is quicker)

Comment (aminalaee) [reply src/picologging/__init__.py:29]: Yes you're right, but it's only used in the tests.
And the reason I dide expose it was that logging module does this.
Maybe we can just import it in the tests and avoid it here.

Comment (aminalaee) [reply src/picologging/__init__.py:29]: Or we can just create this in `picologging` it's easier.

Comment (aminalaee) [reply src/picologging/__init__.py:29]: Done.

**Pull request merged**